vignettes/Model.Rmd

```diff
@@ -10,90 +12,365 @@ vignette: >
 
-The mosquito biological processes are spread out between the following files:
+The `get_parameters()` function generates a complete parameter set that may be fed into `run_simulation()`. A number of **helper functions** have been designed to assist in changing and setting key parameters, which are explained across the remaining vignettes.
 
- 1. 'src/mosquito_ode.cpp' (green)
- 2. 'src/mosquito_emergence.cpp' (green)
- 3. 'R/mosquito_biology.R' (orange)
- 4. 'R/mosquito_biology.R' (red)
+Some parameters (e.g. population size, age group rendering, setting seasonality) must still be replaced directly. When this is the case, care must be taken to ensure the replacement parameters are in the same class as the default parameters (e.g. if the parameter is a numeric, its replacement must also be numeric, if logical, the replacement must also be logical). Parameters are replaced by passing a list of named parameters to the `get_parameters()` function using the `overrides` argument. An example showing how to change the `human_population` parameter is given below.
 
-```{r echo=FALSE}
-DiagrammeR::grViz("digraph {
-  graph [layout = dot, rankdir = LR]
+```{r}
+# Use get_parameters(overrides = list(...))) to set new parameters
+new_params <- get_parameters(overrides = list(human_population = 200)) 
+```
+
+While other parameters can be changed individually, we do not generally recommended adjusting these without close attention and a detailed understanding of how this will impact the model assumptions. We strongly encourage users to stick with the parameter setting functions and methods described in these vignettes when adjusting parameter settings.
+
+## Mosquito modelling
+
+### Species distribution
+
+`malariasimulation` contains in built parameter sets for three mosquito species: *A. arabiensis* (`arab_params`), *A. funestus* (`fun_params`) and *A. gambiae* (`gamb_params`). Each parameter set contains feeding rates, foraging time etc. Full details can be found in the [vector_parameters.R](https://github.com/mrc-ide/malariasimulation/blob/master/R/vector_parameters.R) file.
+
+```{r}
+unlist(arab_params)
+```
+
+The proportion of each species can be set using the `set_species()` function, requiring a full parameter set and arguments of a list of species parameter sets and a vector of relative species proportions.
+
+```{r, fig.align = 'center', fig.height = 5, fig.width = 7}
+# Create list of default parameters
+params <- get_parameters()
+
+# Update parameter list with species distributions
+params_species <- set_species(parameters = params,
+                              species = list(arab_params, fun_params, gamb_params),
+                              proportions = c(0.1,0.3,0.6))
+
+# Run simulation
+species_simulation <- run_simulation(timesteps = 100, parameters = params_species)
+
+## Plot species distributions
+# Collect species column names
+arab_cols <- paste0(c("E","L","P","Sm","Pm", "Im", "NonExistent"),"_arab_count")
```

Comment:
My document won't render and gets stuck here - NonExistent is not an output of the model in this chunk as well as further down.

Reply:
Ah - I was using the release version, rather than the dev version. I've updated it now.